pkg/repositories/config/migrations.go

```diff
@@ -359,6 +359,28 @@ var Migrations = []*gormigrate.Migration{
 			return alterTableColumnType(db, "id", "int")
 		},
 	},
+
+	// Add state to execution model.
+	{
+		ID: "2022-01-11-execution-state",
+		Migrate: func(tx *gorm.DB) error {
+			return tx.AutoMigrate(&models.Execution{})
+		},
+		Rollback: func(tx *gorm.DB) error {
+			return tx.Table("execution").Migrator().DropColumn(&models.Execution{}, "state")
+		},
+	},
+
+	// Add stateUpdatedAt to execution model
```

Comment:
why do these need to be two separate migrations?

Reply:
NA as i have removed the stateUpdatedAt column now.

The thought was to have simple logic in rollback . 

If we have two columns migrated in the same migration key then we might have to unnecessarily rollback a successful column migration due to failed migration on other columns 
OR we might have to handle those cases in the rollback. 

I think it works well if the columns are related which is true in this case since both are related to state details change, but if they are unrelated then i am not sure if we should fail the migration if one of the column migrations failed in a multicolumn migration

Eg : the two columns could be for two different features , then it would block both the features.
Let me know what you think

Reply:
agreed, since the column additions are part of the same change it makes sense that the migration is "transactional" in this case and both should be added in one shot